easybuild/tools/filetools.py

```diff
@@ -2980,3 +2987,81 @@ def create_unused_dir(parent_folder, name):
     set_gid_sticky_bits(path, recursive=True)
 
     return path
+
+
+def get_first_nonexisting_parent(path):
+    path = os.path.abspath(path)
+
+    first_nonexisting_parent = None
+    while not os.path.exists(path):
+        first_nonexisting_parent = path
+        path = os.path.dirname(path)
+
+    return first_nonexisting_parent
+
+
+def _apply_suffix(paths, number):
+    suffix = '_' + str(number)
+
+    final_paths = []
+    first_nonexisting_parents = []
+    for p in paths:
+        final_paths.append(p + suffix)
+        first_nonexisting_parents.append(p)
+
+    return final_paths, first_nonexisting_parents
+
+
+def _clean_directories(paths):
+    for p in paths:
+        shutil.rmtree(p, ignore_errors=True)
+
+
+def create_unused_dirs(paths, index_upper_bound=10000):
+    """
+    Create directories with given paths, including the parent directories
+    When a directory in the same path for any of the path  in the path list already exists, then the suffix '_<i>' is
+    appended for i=0..(index_upper_bound-1) until an index is found so that all required path as unused. All created
+    directories have the same postfix.
+
+    :param paths: list of paths where directories are created; parent path created as well
+    :param index_upper_bound: maximum index that will be tried before failing
+    """
+    paths = list(map(os.path.abspath, paths))
+    for path in paths:
+        for parent in set(paths) - {path}:
+            if is_predecessor(parent, path):
+                raise EasyBuildError("Path '%s' is predecesor of '%s'" % (parent, path))
+
+    final_paths = paths
+    first_nonexisting_parents = list(map(get_first_nonexisting_parent, paths))
+    paths_exist = False
+    number = -1
+    while number < index_upper_bound and not paths_exist:  # Start with no suffix and limit the number of attempts
+        tried_paths = []
+        if number >= 0:
+            final_paths, first_nonexisting_parents = _apply_suffix(paths, number)
+        try:
+            for p in final_paths:
+                tried_paths.append(p)
+                os.makedirs(p)
+            paths_exist = True
+        except OSError as err:
+            # Distinguish between error due to existing folder and anything else
+            if tried_paths != [] and not os.path.exists(tried_paths[-1]):
```

Comment:
It can never be empty because it checks only why the last folder added to the list couldn't be created, can it?
```suggestion
            if not os.path.exists(tried_paths[-1]):
```

Note: `x != []` can be simply `x`

Reply:
It can be `tried_paths = []` only if an empty list of paths is provided as argument. I wouldn't like the function to fail in this case, it should just do nothing.

Reply:
I agree that the test `tried_paths != []` is super flows, test removed.

I will leave the empty list input as is though. The empty list is effectively an identity element for the function, it may appear naturally in some application.